samza-core/src/main/java/org/apache/samza/config/BlobStoreConfig.java

```diff
@@ -19,23 +19,75 @@
 
 package org.apache.samza.config;
 
+import java.time.temporal.ChronoUnit;
+import org.apache.samza.util.RetryPolicyConfig;
+
+
 /**
  * Config related helper methods for BlobStore.
  */
 public class BlobStoreConfig extends MapConfig {
 
-  public static final String BLOB_STORE_MANAGER_FACTORY = "blob.store.manager.factory";
-  public static final String BLOB_STORE_ADMIN_FACTORY = "blob.store.admin.factory";
+  private static final String PREFIX = "blob.store.";
+  public static final String BLOB_STORE_MANAGER_FACTORY = PREFIX + "manager.factory";
+  public static final String BLOB_STORE_ADMIN_FACTORY = PREFIX + "admin.factory";
+  // Configs related to retry policy of blob stores
+  private static final String RETRY_POLICY_PREFIX = PREFIX + "retry.policy.";
+  public static final String RETRY_POLICY_MAX_RETRIES = RETRY_POLICY_PREFIX + "max.retries";
+  // -1 for RetryPolicy means unlimited retries. Retry is limited by max retry duration, rather than count of retries.
+  public static final int DEFAULT_RETRY_POLICY_MAX_RETRIES = -1;
+  public static final String RETRY_POLICY_MAX_RETRY_DURATION_MILLIS = RETRY_POLICY_PREFIX + "max.retry.duration.millis";
+  public static final long DEFAULT_RETRY_POLICY_MAX_RETRY_DURATION_MILLIS = 15 * 60 * 1000; // 15 mins
+  public static final String RETRY_POLICY_BACKOFF_DELAY_MILLIS = RETRY_POLICY_PREFIX + "backoff.delay.millis";
+  public static final long DEFAULT_RETRY_POLICY_BACKOFF_DELAY_MILLIS = 100;
+  public static final String RETRY_POLICY_BACKOFF_MAX_DELAY_MILLIS = RETRY_POLICY_PREFIX + "backoff.max.delay.millis";
+  public static final long DEFAULT_RETRY_POLICY_BACKOFF_MAX_DELAY_MILLIS = 312500;
```

Comment:
Out of curiosity why did we choose this number? Maybe some documentation on how we arrived at this

Reply:
This is a little over 5 mins - the retries with backoff should result in 10 mins retry with this max retry delay (for the last retry).